src/torcontrol.cpp

```diff
@@ -323,6 +376,10 @@ static std::pair<bool,std::string> ReadBinaryFile(const fs::path &filename, size
     char buffer[128];
     size_t n;
     while ((n=fread(buffer, 1, sizeof(buffer), f)) > 0) {
+        // Check for reading errors so we don't return any data if we couldn't
+        // read the entire file (or up to maxsize)
+        if (ferror(f))
+            return std::make_pair(false,"");
```

Comment:
No, there is [no such guarantee](http://www.cplusplus.com/reference/cstdio/fread/#return):

> If [the return value] differs from the _count_ parameter, either a reading error occurred or the _end-of-file_ was reached while reading. In both cases, the proper indicator is set, which can be checked with ferror and feof, respectively.

So it is entirely possible for `fread` to return a number `0 < retval < count` if a reading error occurs.